### PR TITLE
fix: Disable CI/CD linter check with rand library, fix server overlap, and data race issue

### LINKED ISSUE #910: rollkit/p2p: unit test failure with data race flag
Found during rebasing #903 

Reproduction rate: seldom  

```sh
==================
WARNING: DATA RACE
Read at 0x00c0058a9d83 by goroutine 2931:
  testing.(*common).logDepth()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:992 +0xc4
  testing.(*common).log()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:985 +0x84
  testing.(*common).Log()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1026 +0x5b
  github.com/rollkit/rollkit/log/test.(*Logger).Error()
      /home/runner/work/rollkit/rollkit/log/test/loggers.go:46 +0x216
  github.com/rollkit/rollkit/p2p.(*Gossiper).ProcessMessages()
      /home/runner/work/rollkit/rollkit/p2p/gossip.go:96 +0x16c
  github.com/rollkit/rollkit/p2p.(*Client).setupGossiping.func1()
      /home/runner/work/rollkit/rollkit/p2p/client.go:389 +0x58

Previous write at 0x00c0058a9d83 by goroutine 2682:
  testing.tRunner.func1()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1563 +0x82d
  runtime.deferreturn()
      /opt/hostedtoolcache/go/1.20.3/x64/src/runtime/panic.go:476 +0x32
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1629 +0x47

Goroutine 2931 (running) created at:
  github.com/rollkit/rollkit/p2p.(*Client).setupGossiping()
      /home/runner/work/rollkit/rollkit/p2p/client.go:389 +0x3e9
  github.com/rollkit/rollkit/p2p.(*Client).startWithHost()
      /home/runner/work/rollkit/rollkit/p2p/client.go:140 +0x6d6
  github.com/rollkit/rollkit/p2p.startTestNetwork()
      /home/runner/work/rollkit/rollkit/p2p/utils_test.go:117 +0xb84
  github.com/rollkit/rollkit/p2p.TestGossiping()
      /home/runner/work/rollkit/rollkit/p2p/client_test.go:133 +0x9ba
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1629 +0x47

Goroutine 2682 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1629 +0x805
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:2036 +0x8d
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1576 +0x216
  testing.runTests()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:2034 +0x87c
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.20.3/x64/src/testing/testing.go:1906 +0xb44
  main.main()
      _testmain.go:55 +0x2e9
==================
FAIL
FAIL	github.com/rollkit/rollkit/p2p	2.162s
```

### LINKED ISSUE #911: Disable CI/CD linter check with rand library
Current CI/CD linter throws the following error:
`Error: G404: Use of weak random number generator (math/rand instead of crypto/rand) (gosec)`

Need to resolve this.

### LINKED ISSUE #919: Refactor DA tests using TestMain to avoid server overlap
Currently, `da_test.go` tests all individually attempt to start mock `grpc` server and mock `celestia node` servers but those tests use the same port and if those tests are tried to run in parallel, we run into a port already used issue.  Using `TestMain` (as described in https://ieftimov.com/posts/testing-in-go-subtests/) is one way to explore going about it.

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
Resolves #911, resolves #919, resolves #910

### CODE REVIEW

**Comment by smuu** [comment]
In my opinion, the PR title and later the commit message should not contain "Resolve" because this PR does not resolve the security warning, it just disables the check.
I'm okay with not fixing it because it's used in the tests. But this should not be used in production code.

**Comment by Manav-Aggarwal** [comment]
> In my opinion, the PR title and later the commit message should not contain "Resolve" because this PR does not resolve the security warning, it just disables the check. I'm okay with not fixing it because it's used in the tests. But this should not be used in production code.

Since the `math/rand` library is used across multiple files, there's not just one place where we can put a warning. Where do we think is it most appropriate to document the `math/rand` warning if we don't want it to be used in production?

**Review by smuu — DISMISSED** [review]
LGTM

**Review by nashqueue — DISMISSED** [review]

**Review by smuu — APPROVED** [review]

File: p2p/client.go [hunk 1]
```diff
@@ -382,13 +385,14 @@ func (c *Client) setupGossiping(ctx context.Context) error {
 		return err
 	}
 
-	c.txGossiper, err = NewGossiper(c.host, c.ps, c.getTxTopic(), c.logger, WithValidator(c.txValidator))
+	logger := tmlog.NewTMLogger(os.Stdout)
```
Comment (gupadhyaya) [inline p2p/client.go:388]: instead of `tmlog.NewTMLogger` does `logger.With("module", "tx-gossip")` works?

Comment (Manav-Aggarwal) [reply p2p/client.go:388]: added the change but it's not exactly the same. Which logger object would you use for `With`?

Comment (gupadhyaya) [reply p2p/client.go:388]: just use `c.logger.With`?

Comment (Manav-Aggarwal) [reply p2p/client.go:388]: That would reintroduce the problem. Not using `c.logger` is the fix for the issue: https://github.com/rollkit/rollkit/issues/910#issuecomment-1535788972

**Review by nashqueue — DISMISSED** [review]

**Review by gupadhyaya — APPROVED** [review]

**Pull request merged**